components/sup/src/manager/service/supervisor.rs

```diff
@@ -122,23 +124,30 @@ impl Supervisor {
 
     // NOTE: the &self argument is only used to get access to
     // self.preamble, and even then only for Linux :/
-    #[cfg(target_os = "linux")]
+    #[cfg(unix)]
```

Comment:
I made this change to allow `cargo check` to work on macos once again. It's not theoretically the right thing, but as you point out the things below it weren't right either. I can swap it back but it will break `cargo check` on macos